src/RestApi.php

```diff
@@ -61,19 +61,19 @@ public static function get_routes_from_namespace( $namespace ) {
 	 *
 	 * @todo check compat < WC 3.6. Make specific to cart endpoint.
 	 * @param mixed $return Value being filtered.
-	 * @param array $request Request data.
 	 * @return mixed
 	 */
-	public static function maybe_init_cart_session( $return, $request = false ) {
-		// Pass through other errors.
-		if ( ! empty( $error ) ) {
-			return $error;
+	public static function maybe_init_cart_session( $return ) {
+		$wc_instance = wc();
+		// if WooCommerce instance isn't available or already have an
+		// authentication error, just return.
+		if ( ! $wc_instance instanceof \WooCommerce || \is_wp_error( $return ) ) {
```

Comment:
I think the wp error check here is a good call; I can't see any immediate problem here because auth would fail if a WP_Error was present 👍